beprof/curve.py

```diff
@@ -66,19 +66,74 @@ def y_at_x(self, x):
             return x
         return np.interp(x, self.x, self.y, left=np.nan, right=np.nan)
 
+    def change_domain(self, domain):
+
+        # check if new domain includes in the orginal domain
+        if np.max(domain) > np.max(self.x):
```

Comment:
I can't see the point. I am not testing whether one number is between a pair of 2, but I am comaring 2 numbers with first if statement and then another 2 numbers with another if. Where should I put this comparision you mentioned?